src/plugins/identity/README.md

```diff
@@ -1,19 +1,144 @@
+
+Code-reviewed support for Grain built into SourceCred should handle this in
+the future. But for now, requires the instance maintainer to handle Grain
+account contraction.
+
+## Configuration
+
+Identities can be configured using the `project.json` file.
+By adding entries to the the `identities` array.
+
+Below is an example `project.json` file which defines two identities, "User-A"
+and "User-B", contracting their respective GitHub and Discourse accounts.
+
+```js
+[
+  {
+    "type": "sourcecred/project",
+    "version": "0.4.0",
+  },
+  {
+    // The identities we should contract.
+    "identities": [
+      {
+        "username": "User-A",
+        "aliases": ["github/user-a", "discourse/user-a"],
+      },
+      {
+        "username": "User-B",
+        "aliases": ["github/user-b", "discourse/user-b"],
+      },
+    ],
+
+    "id": "@example-project",
+    "discourseServer": {
+      // ...
+    },
+    "repoIds": [
+      // ...
+    ],
+  },
+]
+```
+
+### Username
+
+The `username` field allows you to choose a new username for the contracted
+identity.
+
+It's case-sensitive, must be _unique in this project_ and can contain `A-Z`,
+`a-z`, `0-9`, `-` or `_` as valid characters.
+
+For display purposes the username is prefixed with an `@`.
+If a username is prefixed with `@` in the project file, it will be ignored.
+Meaning `"username": "User-A"` and `"username": "@User-A"` are equivalent.
+
+### Aliases
+
+Aliases are strings with a format of: `{plugin}/{user ID for this plugin}`
+Currently supported plugins are:
+
+| Plugin    | Example alias            | Notes                                                              |
+|:----------|:-------------------------|:-------------------------------------------------------------------|
+| GitHub    | `github/example-user`    | A user account on GitHub's platform.                               |
+| Discourse | `discourse/example-user` | A user account on the Discourse server configured in this project. |
+
+## How a contraction works
+
+Here is a simplified illustration of how contracting works under the hood.
+
+Note that contractions are done _after_ the plugins have created their
+respective parts of the graph. But _before_ any Cred scores are calculated.
+
+Let’s assume we have a user `abc`, who has both GitHub and Discourse accounts
+which we want to contract.
+
+The GitHub and Discourse plugins have mapped out the contributions, users and
+edges as they exist on those services. Even though we merge this into a single
+graph, the GitHub and Discourse contributions of the _same person_ are still
+disconnected.
+
+```
+GitHub user @abc
+├── authored Pull Request #123
+└── authored Issue #456
+
+Discourse user /u/abc
+├── authored Post /t/123/1
+└── authored Post /t/123/2
+```
+
+The identity plugin creates a new node, representing the Identity, with the
+username provided in the project file. While removing the alias nodes.
```

Comment:
The grammar nazi in me cannot let `. While` go. I think it works with `, while`, but that leaves a lot of commas... I think we can make this sentence less choppy by removing a couple of the other commas, i.e.,

> The identity plugin creates a new node representing the Identity with the
> username provided in the project file, while removing the alias nodes.

Reply:
I see what you mean. Mostly this part about the project file is throwing a wrench into this. What about:

> Using the information from the project file, the plugin creates a new Identity node and removes the alias nodes.

Reply:
Works for me

Reply:
Added this and now merged :D